eng/build.yml

```diff
@@ -320,3 +321,168 @@ jobs:
+          - _SignArgs: /p:DotNetSignType=$(_SignType) /p:TeamName=$(_TeamName)
+          - _PerfIterations: 25
+
+      steps:
+      - ${{ if ne(variables['System.TeamProject'], 'public') }}:
+        - task: PowerShell@2
+          displayName: Setup Private Feeds Credentials
+          inputs:
+            filePath: $(Build.SourcesDirectory)/eng/common/SetupNugetSources.ps1
+            arguments: -ConfigFile $(Build.SourcesDirectory)/NuGet.config -Password $Env:Token
+          env:
+            Token: $(dn-bot-dnceng-artifact-feeds-rw)
+      - ${{ if contains(parameters.agentOs, 'Windows_NT') }}:
+        - script: eng\CIBuild.cmd
+                    -configuration $(_BuildConfig)
+                    $(_PublishArgs)
+                    $(_SignArgs)
+                    $(_OfficialBuildIdArgs)
+                    /p:Test=false
+                    $(_InternalRuntimeDownloadArgs)
+          displayName: Build
+          env:
+            BuildConfig: $(_BuildConfig)
+            SYSTEM_ACCESSTOKEN: $(System.AccessToken)
+            HelixAccessToken: ${{ parameters.HelixAccessToken }}
+        - powershell: eng\common\build.ps1
+                -configuration $(_BuildConfig)
+                -prepareMachine
+                -ci
+                -restore
+                $(_Test)
+                -projects $(Build.SourcesDirectory)\src\Tests\dotnet-new.Tests\dotnet-new.IntegrationTests.csproj
+                /bl:$(Build.SourcesDirectory)\artifacts\log\$(_BuildConfig)\dotnet-new.IntegrationTests.binlog
+                $(_InternalRuntimeDownloadArgs)
+          displayName: Run dotnet new Integration Tests
+          condition: succeededOrFailed()
+        - powershell: eng\common\build.ps1
+                -configuration $(_BuildConfig)
+                -prepareMachine
+                -ci
+                -restore
+                $(_Test)
+                -projects $(Build.SourcesDirectory)\src\Tests\Microsoft.TemplateEngine.Cli.IntegrationTests\Microsoft.TemplateEngine.Cli.IntegrationTests.csproj
+                /bl:$(Build.SourcesDirectory)\artifacts\log\$(_BuildConfig)\TemplateEngine.Cli.IntegrationTests.binlog
+                $(_InternalRuntimeDownloadArgs)
+          displayName: Run TemplateEngine Cli Integration Tests
+          condition: succeededOrFailed()
+        - powershell: eng\common\build.ps1
+                -configuration $(_BuildConfig)
+                -prepareMachine
+                -ci
+                -restore
+                $(_Test)
+                -projects $(Build.SourcesDirectory)\src\Tests\Microsoft.TemplateEngine.Cli.UnitTests\Microsoft.TemplateEngine.Cli.UnitTests.csproj
+                /bl:$(Build.SourcesDirectory)\artifacts\log\$(_BuildConfig)\TemplateEngine.Cli.UnitTests.binlog
+                $(_InternalRuntimeDownloadArgs)
+          displayName: Run TemplateEngine Cli Unit Tests
+          condition: succeededOrFailed()
```

Comment:
nit: this project may be run together with other jobs. We can move it in different PR.

Reply:
Unit tests should be run prior to integration tests